modules/hal_nordic/nrfx/Kconfig

```diff
@@ -76,172 +74,168 @@ config NRFX_GPIOTE_NUM_OF_EVT_HANDLERS
 
 config NRFX_I2S
 	bool "I2S driver"
-	depends on HAS_HW_NRF_I2S
+	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_I2S))
 
 config NRFX_IPC
 	bool "IPC driver"
-	depends on HAS_HW_NRF_IPC
+	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_IPC))
 
 config NRFX_LPCOMP
 	bool "LPCOMP driver"
-	depends on HAS_HW_NRF_LPCOMP
+	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_LPCOMP))
 
 config NRFX_NFCT
 	bool "NFCT driver"
-	depends on HAS_HW_NRF_NFCT
+	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_NFCT))
 	select NRFX_TIMER4 if SOC_SERIES_NRF52X
 	select NRFX_TIMER2 if SOC_SERIES_NRF53X
 
 config NRFX_NVMC
 	bool "NVMC driver"
+	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF51_FLASH_CONTROLLER)) \
+		|| $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF52_FLASH_CONTROLLER)) \
+		|| $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF53_FLASH_CONTROLLER)) \
+		|| $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF91_FLASH_CONTROLLER))
 
 config NRFX_PDM
 	bool "PDM driver"
-	depends on HAS_HW_NRF_PDM
+	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_PDM))
 
 config NRFX_POWER
 	bool "POWER driver"
-	depends on HAS_HW_NRF_POWER
+	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_POWER))
 	# On SoCs featuring the USBREG peripheral, the POWER driver uses
 	# internally the USBREG driver.
-	select NRFX_USBREG if HAS_HW_NRF_USBREG
+	select NRFX_USBREG if $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_USBREG))
 
 config NRFX_PPI
 	bool "PPI allocator"
-	depends on HAS_HW_NRF_PPI
+	# depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_PPI))
```

Comment:
I guess this is commented out because it is still a draft?

Reply:
To some extent. :wink: I actually forgot to restore this dependency. Thanks for spotting it!